app/domain/authentication/authn_jwt/authenticator.rb

```diff
@@ -41,11 +50,65 @@ def get_jwt_identity
         @authentication_parameters.jwt_identity =  @jwt_configuration.jwt_identity(@authentication_parameters)
       end
 
+      def validate_origin
+        @validate_origin.(
+          account: @authentication_parameters.account,
+          username: @authentication_parameters.username,
+          client_ip: @authentication_parameters.client_ip
+        )
+      end
+
       def validate_restrictions
         @logger.debug(LogMessages::Authentication::AuthnJwt::CALLING_VALIDATE_RESTRICTIONS.new)
         @jwt_configuration.validate_restrictions(@authentication_parameters)
       end
 
+      def audit_success
+        @audit_log.log(
+          ::Audit::Event::Authn::Authenticate.new(
+            authenticator_name: @authentication_parameters.authenticator_name,
+            service: webservice,
+            role_id: audit_role_id,
+            client_ip: @authentication_parameters.client_ip,
+            success: true,
+            error_message: nil
+          )
+        )
+      end
+
+      def audit_failure(err)
+        @audit_log.log(
+          ::Audit::Event::Authn::Authenticate.new(
+            authenticator_name: @authentication_parameters.authenticator_name,
+            service: webservice,
+            role_id: audit_role_id,
```

Comment:
Another flow bug is there.
Role will be knows only after jwt signature validation until this moment it's undefined.